crates/iceberg-catalog/src/catalog/io.rs

```diff
@@ -1,14 +1,110 @@
 use crate::api::{ErrorModel, Result};
 use crate::service::storage::path_utils;
 use flate2::{write::GzEncoder, Compression};
-use iceberg::io::FileIO;
+use iceberg::{io::FileIO, spec::view_properties::METADATA_COMPRESSION};
 use iceberg_ext::catalog::rest::IcebergErrorResponse;
 use serde::Serialize;
+use std::collections::HashMap;
 use std::io::Write;
 
+use super::CommonMetadata;
+
+#[derive(thiserror::Error, Debug, strum::IntoStaticStr)]
+pub enum CompressionCodecError {
+    #[error("Unsupported compression codec: {0}")]
+    UnsupportedCompressionCodec(String),
+}
+
+impl CompressionCodecError {
+    pub fn to_type(&self) -> &'static str {
+        self.into()
+    }
+}
+
+impl From<CompressionCodecError> for IcebergErrorResponse {
+    fn from(value: CompressionCodecError) -> Self {
+        let typ = value.to_type();
+        let boxed = Box::new(value);
+        let message = boxed.to_string();
+
+        match value {
+            CompressionCodecError::UnsupportedCompressionCodec(_) => {
+                ErrorModel::bad_request(message, typ, Some(boxed)).into()
+            }
+        }
+    }
+}
```

Comment:
I don't expect other variants. I made it an enum because I wanted to mimic the existing code to create an `IcebergErrorResponse`, which was necessary because `CompressionCodec` is used in places where an `IcebergErrorResponse` is expected. But you are of course correct, a `struct` is more appropriate, and an `IcebergErrorResponse` can still be created.